app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt

```diff
@@ -24,6 +27,57 @@ data class Wallpaper(
  * A type hierarchy representing the different theme collections [Wallpaper]s belong to.
  */
 sealed class WallpaperThemeCollection {
-    object None : WallpaperThemeCollection()
-    object Firefox : WallpaperThemeCollection()
+    abstract val origin: WallpaperOrigin
+    object None : WallpaperThemeCollection() {
+        override val origin: WallpaperOrigin = WallpaperOrigin.Local
+    }
+    object Firefox : WallpaperThemeCollection() {
+        override val origin: WallpaperOrigin = WallpaperOrigin.Local
+    }
+    object Focus : WallpaperThemeCollection() {
+        override val origin: WallpaperOrigin = WallpaperOrigin.Remote
+    }
+}
+
+/**
+ * Get the parent directory name for a remote wallpaper asset.
+ */
+fun WallpaperThemeCollection.directoryName(): String = when (this) {
+    WallpaperThemeCollection.None,
+    WallpaperThemeCollection.Firefox -> ""
+    WallpaperThemeCollection.Focus -> "focus"
+}
+
+/**
+ * Types defining whether a [Wallpaper] is delivered through a remote source or is included locally
+ * in the APK.
+ */
+sealed class WallpaperOrigin {
+    object Local : WallpaperOrigin()
+    object Remote : WallpaperOrigin()
+}
```

Comment:
Oh yeah, I like that! We could even designate them as `Local` or `Remote` as part of the type hierarchy. Something like:

```
sealed class Wallpaper(...) {
    sealed class Local(...) : Wallpaper(...){
        data class FirefoxWallpaper(...) : Local(...)
        ...
    }
}
```

But what if we refactor it to use the `State` pattern eventually? I'm not immediately finding examples of any `State` objects that have `sealed class` properties. A `sealed class` has the same issue that an `enum class` does, in that we could define an `object` child with global mutable state, something like 

```
sealed class Wallpaper(...) {
    sealed class Local(...) : Wallpaper(...){
        object Default : Local {
            var resourceId: Int // changes based on app light or dark theme
        ...
    }
}
```

It seemed like we wanted to avoid using anything but nested `data class`es for state, when this came up in my first attempt at the refactor in #23113. Are `sealed class`es okay, but not `enum class`es? What do you think @gabrielluong ?

Reply:
My conclusion back then was that they are both fined. The main gotcha is what I presented in our slack discussion. Ideally, we just don't have too much functionality within the class.

Reply:
Okay cool. I'll update this in the follow-up PR that moves the wallpapers from `asset`s to `drawable`s, as it'll be easier when not relying on the dynamic data contained in `wallpapers.json`.